### PR TITLE
Added actions, reducers for books and categories and common state

### PR DESCRIPTION
**Pull request by Anand8317**
- Added action (add and remove), action creators, and reducers for books.
- Added action (check status), action creators, and reducers for categories.
- Added a store with the above two reducers combined to root reducer.

### CODE REVIEW

File: src/redux/categories/categories.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const CHECKSTATUS = 'my-app/categories/CHECKSTATUS';
```
Comment (gemmen29) [inline src/redux/categories/categories.js]: - [ ] Kindly update the action type to 
```javascript 
const CHECKSTATUS = 'bookstore/categories/CHECKSTATUS';
```
to match the duck pattern guidelines.

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi @Anand8317, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/categories/categories.js (hunk 1 above)
Comment (Anand8317) [reply src/redux/categories/categories.js]: Thank you for pointing this out. I have updated it now.

**Review by AnselemOdims — APPROVED** [review]
### Hi  @Anand8317 🙋‍♀️, 
Good job in making the required changes 👏 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Use the tag @AnselemOdims for a quicker reply


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by nahnahdev — COMMENTED** [review]
## Project Status: APPROVED :green_circle: 
Hi @Anand8317 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights
- All required changes met :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**